apps/voting/contracts/Voting.sol

```diff
@@ -104,6 +122,79 @@ contract Voting is IForwarder, AragonApp {
         objectionPhaseTime = _objectionPhaseTime;
     }
 
+    /**
+     * TODO: Calculate gas spending and add hint for more efficient look up in the array if needed
+     */
+    function setVotingManager(address _manager) public {
+        require(_manager != address(0), ERROR_ZERO_ADDRESS_PASSED);
+
+        address msgSender = msg.sender;
+        require(_manager != msg.sender, ERROR_SELF_MANAGER);
+
+        address prevManager = managers[msgSender];
+        require(_manager != prevManager, ERROR_MANAGER_SAME_AS_PREV);
+
+        if (prevManager != address(0)) {
+            _removeManagedAddressFor(prevManager, msgSender);
+        }
+
+        _addManagedAddressFor(_manager, msgSender);
+        managers[msgSender] = _manager;
+
+        emit ManagerSet(msgSender, prevManager, _manager);
+    }
+
+    /**
+     * TODO: Calculate gas spending and add hint for more efficient look up in the array if needed
+     */
+    function removeVotingManager() public {
+        address msgSender = msg.sender;
+        address prevManager = managers[msgSender];
+        require(prevManager != address(0), ERROR_MANAGER_NOT_SET);
+
+        _removeManagedAddressFor(prevManager, msgSender);
+        managers[msgSender] = address(0);
+
+        emit ManagerSet(msgSender, prevManager, address(0));
+    }
+
+    function _addManagedAddressFor(address _manager, address _holder) public {
+        require(_manager != address(0), ERROR_ZERO_ADDRESS_PASSED);
+        require(_holder != address(0), ERROR_ZERO_ADDRESS_PASSED);
+
+        uint256 length = managedHolders[_manager].addresses.length;
+        require(length < MAX_HOLDERS_PER_MANAGER, ERROR_MAX_MANAGED_VOTERS_REACHED);
```

Comment:
Opens the possibility for the spamming of the managers. Create many accounts and delegate voting power to the same manager. Then, no one can delegate to this manager.